### PR TITLE
Common ginkgo lint

### PR DESCRIPTION
**Pull request by nunnatsa**
Replace the ginkgo-linter from the source code, with a newer version
from https://github.com/nunnatsa/ginkgolinter. This version add two more
validations, in addition to the existing wrong length validation:

* wrong nil assertion; e.g.
   `Expect(x).To(Eaual(nil))` => `Expect(x).To(BeNil()`
   `Expect(x == nil).To(BeTrue())` => `Expect(x).To(BeNil()`
* wrong error assertion; e.g.
   `Expect(err).To(BeNil())` => `Expect(err).ToNot(HaveOccurred())`
   `Expect(funcReturnsError()).To(BeNil())` => `Expect(funcReturnsError()).To(Succeed())`

The reason for taking this linter code out is to ease the development of
it, independatly from the kubevirt development. Also, it will allow
other projects to use it as well.

Also, this PR fixes all the linter new warnings, so it could start working right away.

```release-note
None
```

### CODE REVIEW

**Comment by nunnatsa** [comment]
/test 
pull-kubevirt-e2e-kind-1.23-vgpu-nonroot

**Review by EdDev — COMMENTED** [review]
Thank you!!
As a whole, this PR changes look great.

It was challenging to review so many changes, but I made it :)

Please note that the first commit is also removing and adding some files, which I think it was not intentionally (that was suppose to fall under the 2nd commit I think).

If there will be changes added to this PR, including rebases, please try to:
- Do not mix rebases with changes related to this topic (i.e. push once the rebase and then the fixes needed to adjust to pass the linter). This way we will not need to review everything again and avoid partial commits.

Would you be able to move the file addition/deletion to the 2nd commit?

**Comment by nunnatsa** [comment]
> Would you be able to move the file addition/deletion to the 2nd commit?

Thanks @EdDev - done.

**Review by EdDev — APPROVED** [review]
Thank you very much! Great work!

File: pkg/virt-controller/watch/vm_test.go [hunk 1]
```diff
@@ -985,12 +985,8 @@ var _ = Describe("VirtualMachine", func() {
 				for i := 0; i < 1000; i++ {
 					delay := calculateStartBackoffTime(failCount, defaultMaxCrashLoopBackoffDelaySeconds)
 
-					if delay > maxExpectedDelay {
```
Comment (xpivarc) [inline pkg/virt-controller/watch/vm_test.go:988]: +1

**Review by xpivarc — COMMENTED** [review]
/approve

**Pull request merged**